### PR TITLE
add lomap2 + openfe

### PR DESCRIPTION
**Pull request by mikemhenry**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by mikemhenry** [comment]
@richardjgowers @IAlibay @dwhswenson

Can you all comment here and say that you are willing to maintain this package?
Thanks!

**Comment by dwhswenson** [comment]
I'm willing to be a maintainer

**Comment by richardjgowers** [comment]
Yep!

**Comment by mikemhenry** [comment]
@richardjgowers 
Looks like some issue with windows + rdkit:
```
import: 'lomap'
Matplotlib is building the font cache; this may take a moment.
Traceback (most recent call last):
  File "C:\bld\lomap2_1645726158426\test_tmp\run_test.py", line 2, in <module>
    import lomap
  File "C:\bld\lomap2_1645726158426\_test_env\lib\site-packages\lomap\__init__.py", line 62, in <module>
    from .dbmol import DBMolecules
  File "C:\bld\lomap2_1645726158426\_test_env\lib\site-packages\lomap\dbmol.py", line 38, in <module>
    from . import graphgen
  File "C:\bld\lomap2_1645726158426\_test_env\lib\site-packages\lomap\graphgen.py", line 42, in <module>
    from rdkit.Chem import Draw
  File "C:\bld\lomap2_1645726158426\_test_env\lib\site-packages\rdkit\__init__.py", line 2, in <module>
    from .rdBase import rdkitVersion as __version__
ImportError: DLL load failed: The specified procedure could not be found.

(%PREFIX%) %SRC_DIR%>IF 1 NEQ 0 exit /B 1 
Tests failed for lomap2-2.0.0_rc-py37h7fedcae_0.tar.bz2 - moving package to C:\bld\broken
WARNING:conda_build.build:Tests failed for lomap2-2.0.0_rc-py37h7fedcae_0.tar.bz2 - moving package to C:\bld\broken
WARNING conda_build.build:tests_failed(2975): Tests failed for lomap2-2.0.0_rc-py37h7fedcae_0.tar.bz2 - moving package to C:\bld\broken
TESTS FAILED: lomap2-2.0.0_rc-py37h7fedcae_0.tar.bz2
```
Do we care about having a windows lomap package?

**Comment by richardjgowers** [comment]
Nah no windows for now

**Comment by mikemhenry** [comment]
@conda-forge/staged-recipes this is ready for review! Our fork is going to be the official source as we are taking over the project from https://github.com/MobleyLab/Lomap

**Comment by isuruf** [comment]
> Our fork is going to be the official source as we are taking over the project from https://github.com/MobleyLab/Lomap

What does this mean?

**Comment by dwhswenson** [comment]
@isuruf : There's a plan to detach the fork relationship between OpenFreeEnergy/Lomap and the older MobleyLab/Lomap repo. I think this will also involve archiving the MobleyLab version. This was chosen over transferring the MobleyLab version to the OpenFreeEnergy org.

There currently is no PyPI or conda-forge source for this lomap. (There is a completely unrelated package called lomap on PyPI; we're lomap2 partly to avoid name conflicts).

**Comment by mikemhenry** [comment]
@davidlmobley can confirm that we are taking over lomap development. I only mention it because of the checklist:
>  Source is from official source.

**Comment by davidlmobley** [comment]
Yes, I confirm that my lab is handing over long-term maintenance of LOMAP(2) to the Open Free Energy project, so the MobleyLab/Lomap repo will no longer be the authoritative version and theirs will be.

File: recipes/lomap2/meta.yaml
Comment (isuruf) [inline recipes/lomap2/meta.yaml:13]: Can you remove this and add `noarch: python`?

File: recipes/lomap2/meta.yaml
Comment (isuruf) [inline recipes/lomap2/meta.yaml:2]: Since this is an rc, can you add a `recipe/conda_build_config.yaml` with 
```yaml
channel_targets:
- conda-forge lomap2_rc
```

**Comment by mikemhenry** [comment]
@isuruf Did I put this in the wrong spot?

`ValueError: Specified folder /home/conda/staged-recipes-copy/recipes/conda_build_config.yaml does not exist`
https://github.com/conda-forge/staged-recipes/pull/18154/commits/ed8d9977dc6c109ec2dad76d6469811ceb3b8336

**Comment by mikemhenry** [comment]
@richardjgowers @IAlibay @dwhswenson
Can you all comment (again) that you are willing to maintain `openfe` 
Thanks!

**Comment by mikemhenry** [comment]
This will be blocked until https://github.com/conda-forge/staged-recipes/pull/18027 is merged

**Comment by IAlibay** [comment]
I confirm that I am happy to be a maintainer

**Comment by dwhswenson** [comment]
I'm willing to be a maintainer for both of these packages.

**Comment by mikemhenry** [comment]
@conda-forge/staged-recipes ready for review!

**Comment by mikemhenry** [comment]
Just a friendly bump :heart: 
@conda-forge/staged-recipes ready for review!

**Comment by richardjgowers** [comment]
@mikemhenry I think we're not supporting Windows because of deps also not, so that will have to get removed (definitely from openfe, maybe from lomap too)

**Comment by mikemhenry** [comment]
@richardjgowers lomap2 can build on windows but openfe needs the openff-toolkit which needs ambertools which isn't on windows, so that is why the CI is failing.

However, these are `noarch: python` packages, from From: https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-builds

> A package is still considered noarch even if one of its dependencies is not available on a given platform. If this is the case, conda will display a helpful error message describing which dependency couldn’t be found when it tries to install the package.

So we should be fine.

**Comment by mikemhenry** [comment]
Ready for review :)
@conda-forge/staged-recipes

File: recipes/lomap2/meta.yaml
Comment (BastianZim) [inline recipes/lomap2/meta.yaml:2]: This needs to target the `dev` label: https://conda-forge.org/docs/maintainer/knowledge_base.html#pre-release-builds

Although it might be easier to just wait until the full version is released.

**Review by BastianZim — CHANGES_REQUESTED** [review]

File: recipes/lomap2/meta.yaml
Comment (mikemhenry) [reply recipes/lomap2/meta.yaml:2]: I had that in this commit 366d72819d540ab7fb7d8f8db9089a9a188a06a1

But will that cause issues with the openfe package? Will conda-build be able to figure that out? 

Thanks so much for your review!

Comment (BastianZim) [reply recipes/lomap2/meta.yaml:2]: Yes, that can be done: https://conda-forge.org/docs/maintainer/knowledge_base.html#creating-a-pre-release-build

But to be honest, it's probably simpler if you make a real release and use that. Since you are upstream as well, I would evaluate if that's possible.

File: recipes/openfe/meta.yaml
Comment (BastianZim) [inline recipes/openfe/meta.yaml:15]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv
```

File: recipes/openfe/meta.yaml
Comment (BastianZim) [inline recipes/openfe/meta.yaml:13]: You'll need to add the entry points here as well.

Ref:

* https://github.com/OpenFreeEnergy/openfe/blob/03cf8a1ca3db5af90a6220742dc2d105173cc0e6/setup.cfg#L41-L43
* https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#python-entry-points

File: recipes/openfe/meta.yaml
Comment (BastianZim) [inline recipes/openfe/meta.yaml:35]: Most of these are not upstream, are you sure?

Ref: https://github.com/OpenFreeEnergy/openfe/blob/03cf8a1ca3db5af90a6220742dc2d105173cc0e6/setup.cfg#L24-L28

**Review by BastianZim — CHANGES_REQUESTED** [review]
`openfe` looks good, just a couple of comments. Then that is ready to go.

I have not yet reviewed `lomap2` until you decide what to do there.

File: recipes/lomap2/meta.yaml
Comment (richardjgowers) [reply recipes/lomap2/meta.yaml:2]: I've cut the full release of Lomap now, so we can target v2.0.0 proper

File: recipes/openfe/meta.yaml
Comment (mikemhenry) [reply recipes/openfe/meta.yaml:17]: Fixed, thanks!

File: recipes/openfe/meta.yaml
Comment (BastianZim) [reply recipes/openfe/meta.yaml]: Ahh ok – perfect!

File: recipes/openfe/meta.yaml
Comment (BastianZim) [inline recipes/openfe/meta.yaml:16]:
```suggestion
    - openfe = openfecli.cli:main
```

File: recipes/openfe/meta.yaml
Comment (BastianZim) [inline recipes/openfe/meta.yaml:17]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv
```

File: recipes/openfe/meta.yaml
Comment (BastianZim) [inline recipes/openfe/meta.yaml:43]:
```suggestion
    - pip check
    - openfe --help
```
This checks that the entry points are actually working. If `--help` is not available in the CLI, feel free to change it to something else.

File: recipes/lomap2/meta.yaml
Comment (BastianZim) [inline recipes/lomap2/meta.yaml:16]:
```suggestion
    - lomap = lomap.dbmol:startup
```

File: recipes/lomap2/meta.yaml
Comment (BastianZim) [inline recipes/lomap2/meta.yaml:36]:
```suggestion
    - pip check
    - lomap --help
```
Just to check the entry points.

**Review by BastianZim — CHANGES_REQUESTED** [review]
Just a couple of small things, otherwise this should be good to go!

**Comment by mikemhenry** [comment]
I can't batch commit these so I will add them manually to reduce CI churn

**Comment by BastianZim** [comment]
That’s ok. The CI will stop when you submit the next one.

**Comment by mikemhenry** [comment]
Cool, I didn't know if that concurrency option was setup on azure, good to know!

**Review by BastianZim — APPROVED** [review]

**Comment by mikemhenry** [comment]
@BastianZim Thanks for the suggestions on testing the entry points!

**Pull request merged**
